luigi/task.py

```diff
@@ -289,6 +289,9 @@ def __init__(self, *args, **kwargs):
         self.task_id = task_id_str(self.task_family, self.to_str_params(only_significant=True))
         self.__hash = hash(self.task_id)
 
+        self.status_message = None
+        self._status_message_callback = None
```

Comment:
I don't get this pair. Shouldn't `status_message` be prefixed with `_` and not the callback? The user is meant to write a callback method, not tamper with the internal  `status_message` variable that's used as a cache.

In fact, why not just remove `status_message`. Are you sure you need the only-post-if-changed logic?